nixos/modules/services/misc/gitea.nix

```diff
@@ -0,0 +1,405 @@
+          description = ''
+            Require users to login to view any page.
+          '';
+        };
+      };
+
+      mail = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Enable mail delivery using SMTP.";
+        };
+
+        host = mkOption {
+          type = types.string;
+          default = "localhost";
+          description = "The host to use for sending mails.";
+        };
+
+        port = mkOption {
+          type = types.int;
+          default = 25;
+          description = "The port to use for the SMTP server.";
+        };
+
+        user = mkOption {
+          type = types.string;
+          default = "";
+          description = "The username to use for the SMTP server.";
+        };
+
+        password = mkOption {
+          type = types.string;
+          default = "";
+          description = "The password to use for the SMTP server.";
+        };
+
+        from = mkOption {
+          type = types.string;
+          default = "gitea@localhost";
+          description = "The email address to use for sending mails.";
+        };
+      };
+
+      footer = {
+        branding = mkOption {
+          type = types.bool;
+          default = true;
+          description = "Show branding in footer.";
+        };
+
+        version = mkOption {
+          type = types.bool;
+          default = true;
+          description = "Show version in footer.";
+        };
+
+        loadtime = mkOption {
+          type = types.bool;
+          default = true;
+          description = "Show loadtime in footer.";
+        };
+      };
+
+      extraConfig = mkOption {
+        type = types.str;
+        default = "";
+        description = "Configuration lines appended to the generated Gitea configuration file.";
+      };
+
+      stateDir = mkOption {
+        default = "/var/lib/gitea";
+        type = types.str;
+        description = "Gitea data directory.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = giteaUser;
+        description = "User account under which Gitea runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = giteaGroup;
+        description = "Group account under which Gitea runs.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.extraUsers = optional (cfg.user == giteaUser) {
+      name = giteaUser;
+      uid = config.ids.uids.gitea;
+      group = cfg.group;
+      description = "Go Git Service";
+      home = cfg.stateDir;
+      createHome = true;
+      shell = pkgs.bash;
+    };
+
+    users.extraGroups = optional (cfg.group == giteaGroup) {
+      name = giteaGroup;
+      gid = config.ids.gids.gitea;
+    };
+
+    services.postgresql = mkIf useLocalPostgresqlDatabase {
+      enable = true;
+    };
+
+    system.activationScripts.gitea = ''
+      mkdir -p ${cfg.stateDir}
+      chown -R ${cfg.user}:${cfg.group} ${cfg.stateDir}
```

Comment:
This might be an expensive operation (if a lot git repositories are present). Since you have allocated static uid/gid number, please only change permission of the root state directory:

```
install -d -g ${cfg.group} -o ${cfg.user} ${cfg.stateDir}
```